src/test/java/io/jenkins/plugins/analysis/core/scm/GitBlameITest.java

```diff
@@ -0,0 +1,219 @@
+        assertThat(analysisResult).hasTotalSize(1);
+
+        HtmlPage detailsPage = getDetailsWebPage(project, analysisResult);
+        SourceControlTable sourceControlTable = new SourceControlTable(detailsPage);
+
+        List<SourceControlRow> sourceControlRows = sourceControlTable.getRows();
+        assertThat(sourceControlRows.get(0).getValue(SourceControlRow.AUTHOR)).isEqualTo("Jane Doe");
+        assertThat(sourceControlRows.get(0).getValue(SourceControlRow.EMAIL)).isEqualTo("Jane@localhost");
+        assertThat(sourceControlRows.get(0).getValue(SourceControlRow.FILE)).isEqualTo(fileName + ":4");
+        assertThat(sourceControlRows.get(0).getValue(SourceControlRow.DETAILS_CONTENT)).isEqualTo(
+                "Unexpected end of File");
+    }
+
+    /**
+     * Checks if git blame issues are generated properly if default checkout is disabled.
+     *
+     * @throws Exception
+     *         If the git commands used are failing.
+     */
+    @Issue("JENKINS-57260")
+    @Test
+    public void shouldShowGitBlameWarningWithDisabledDefaultCheckout() throws Exception {
+        gitRepo.init();
+        appendTextToFileInScm(gitRepo, "Hello.java", "class Hello {}", "John Doe", "John@localhost");
+
+        final String pipeline = "pipeline {\n"
+                + " agent any\n"
+                + " options {\n"
+                + "     skipDefaultCheckout()\n"
+                + " }\n"
+                + " stages {\n"
+                + "     stage('Prepare') {\n"
+                + "         steps {\n"
+                + "             dir('src') {\n"
+                + "                 checkout scm\n"
+                + "                 sh 'ls -ahl'\n"
+                + "             }\n"
+                + "         }\n"
+                + "     }\n"
+                + "     stage('Doxygen') {\n"
+                + "         steps {\n"
+                + "             dir('build/doxygen') {\n"
+                + "                 sh 'mkdir doxygen'\n"
+                + "                 sh 'echo \"src/Hello.java:1: Error: No Javadoc for Class\" > doxygen/doxygen.log'\n"
+                + "             }\n"
+                + "         recordIssues(aggregatingResults: true, enabledForFailure: true, tools: [ doxygen(name: 'Doxygen', pattern: 'build/doxygen/doxygen/doxygen.log') ] )\n"
+                + "         }\n"
+                + "     }\n"
+                + " }\n"
+                + "}";
+
+        WorkflowJob project = createPipeline();
+        appendTextToFileInScm(gitRepo, "Pipeline.txt", pipeline, "Pipeline", "root@localhost");
+        project.setDefinition(new CpsScmFlowDefinition(new GitSCM(gitRepo.toString()), "Pipeline.txt"));
+        AnalysisResult analysisResult = scheduleSuccessfulBuild(project);
+        assertThat(analysisResult).hasTotalSize(1);
+
+        //HtmlPage detailsPage = getDetailsWebPage(project, analysisResult);
```

Comment:
@uhafner In this test the blame issues are not generated properly. We always get the message: "Skipping issues blame since Git is the only supported SCM up to now." Do you know why the git repository is not recognized correctly? The java file which has an error is part of the files checked out at the beginning of the pipeline.

#########################
ls -ahl src
drwxrwxr-x 3 osboxes osboxes 4.0K May 30 02:17 .
drwxrwxr-x 4 osboxes osboxes 4.0K May 30 02:17 ..
-rw-rw-r-- 1 osboxes osboxes    0 May 30 02:17 file
drwxrwxr-x 8 osboxes osboxes 4.0K May 30 02:17 .git
-rw-rw-r-- 1 osboxes osboxes   14 May 30 02:17 Hello.java
-rw-rw-r-- 1 osboxes osboxes  637 May 30 02:17 Pipeline.txt